app/src/screens/Delays.js

```diff
@@ -0,0 +1,200 @@
+    delayAppFilter,
+    handleDelayAppFilterChange,
+    handleClearFilters,
+    executionTargets,
+    selectDelay,
+  }) => {
+    const { layoutName } = useLayout()
+    const theme = useTheme()
+    const { ongoingDelays, pausedDelays, pendingDelays } = useDelays(filteredDelays)
+
+    const multipleOfTarget = executionTargets.reduce((map, { name }) => {
+      map.set(name, map.has(name))
+      return map
+    }, new Map())
+
+    return (
+      <React.Fragment>
+        {layoutName !== 'small' && (
+          <Bar>
+            <div
+              css={`
+                height: ${8 * GU}px;
+                display: grid;
+                grid-template-columns: auto auto auto 1fr;
+                grid-gap: ${1 * GU}px;
+                align-items: center;
+                padding-left: ${3 * GU}px;
+              `}
+            >
+              <DropDown
+                header="Status"
+                placeholder="Status"
+                selected={delayStatusFilter}
+                onChange={handleDelayStatusFilterChange}
+                items={[
+                  <div>
+                    All
+                    <span
+                      css={`
+                        margin-left: ${1.5 * GU}px;
+                        display: inline-flex;
+                        align-items: center;
+                        justify-content: center;
+                        color: ${theme.info};
+                        ${textStyle('label3')};
+                      `}
+                    >
+                      <Tag limitDigits={4} label={delays.length} size="small" />
+                    </span>
+                  </div>,
+                  'Ongoing',
+                  'Paused',
+                  'Pending',
+                ]}
+                width="128px"
+              />
+
+              <DropDown
+                header="App"
+                placeholder="App"
+                selected={delayAppFilter}
+                onChange={handleDelayAppFilterChange}
+                items={[
+                  'All',
+                  <ThisDelay showTag={multipleOfTarget.get('Delay')} />,
+                  ...executionTargets.map(
+                    ({ name, identifier }) =>
+                      `${name}${multipleOfTarget.get(name) && identifier ? ` (${identifier})` : ''}`
+                  ),
+                  'External',
+                ]}
+                width="128px"
+              />
+            </div>
+          </Bar>
+        )}
+
+        <React.Fragment>
+          {!filteredDelays.length ? (
+            <EmptyFilteredDelays onClear={handleClearFilters} />
+          ) : (
+            <DelayGroups
+              ongoingDelays={ongoingDelays}
+              pausedDelays={pausedDelays}
+              pendingDelays={pendingDelays}
+              selectDelay={selectDelay}
+            />
+          )}
+        </React.Fragment>
+      </React.Fragment>
+    )
+  }
+)
+
+const ThisDelay = ({ showTag }) => (
```

Comment:
does it make sense to rename this to something like currentDelay? also i think that we should keep const, variable, and function names starting with lowercase

Reply:
For the renaming i think it makes sense to rename it, though i think a better description could be `ThisApp` as it references the Delay app.

Regarding the lowercase, this are all React components and by convention they are named with Uppercase.

Reply:
ohh didn't notice that they were react components, seems good